src/test/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/AssemblyBasedSVDiscoveryTestDataProvider.java

```diff
@@ -0,0 +1,217 @@
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.SvType;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.alignment.AlignmentInterval;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.utils.GATKSVVCFConstants.*;
+
+public abstract class AssemblyBasedSVDiscoveryTestDataProvider {
+
+    public abstract static class AssemblyBasedSVDiscoveryTestDataForSimpleChimera {
+        // these are used for building
+        public final AlignmentInterval firstAlignment;
+        public final AlignmentInterval secondAlignment;
+        public final String evidenceAssemblyContigName;
+        public final byte[] evidenceContigSeq;
+
+        // these are used for testing
+        public final boolean expectedFirstContigRegionHasLaterReferenceMapping;
+        public final SimpleChimera manuallyCuratedSimpleChimera;
+        public final NovelAdjacencyAndAltHaplotype manuallyCuratedBiPathBubble;
+
+        public final List<SvType> manuallyCuratedSVTypes;
+        public final List<VariantContext> manuallyCuratedVariants;
+
+        public final Class<? extends BreakpointsInference> inferencer;
+
+        public AssemblyBasedSVDiscoveryTestDataForSimpleChimera(final AlignmentInterval firstAlignment, final AlignmentInterval secondAlignment,
+                                                                final String evidenceAssemblyContigName, final byte[] evidenceContigSeq,
+                                                                final boolean expectedFirstContigRegionHasLaterReferenceMapping,
+                                                                final SimpleChimera manuallyCuratedSimpleChimera,
+                                                                final NovelAdjacencyAndAltHaplotype manuallyCuratedBiPathBubble,
+                                                                final List<SvType> manuallyCuratedSVTypes,
+                                                                final List<VariantContext> manuallyCuratedVariants,
+                                                                final Class<? extends BreakpointsInference> inferencer) {
+            this.firstAlignment = firstAlignment;
+            this.secondAlignment = secondAlignment;
+            this.evidenceAssemblyContigName = evidenceAssemblyContigName;
+            this.evidenceContigSeq = evidenceContigSeq;
+            this.expectedFirstContigRegionHasLaterReferenceMapping = expectedFirstContigRegionHasLaterReferenceMapping;
+            this.manuallyCuratedSimpleChimera = manuallyCuratedSimpleChimera;
+            this.manuallyCuratedBiPathBubble = manuallyCuratedBiPathBubble;
+            this.manuallyCuratedSVTypes = manuallyCuratedSVTypes;
+            this.manuallyCuratedVariants = manuallyCuratedVariants;
+            this.inferencer = inferencer;
```

Comment:
Why do you have to store the `inferencer` class along with the test data? Is there any way that you could make selecting the inferencer part of the test too? It seems like it allows the potentially confusing condition of writing a test but then choosing an `inferencer` that wouldn't actually be chosen by the system in an integration setting. Alternatively, why not just have different data providers for each `inferencer` subtype?